ls1intum/Artemis
# `Assessment`: Tooltips for assessment notes

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


https://github.com/ls1intum/Artemis/issues/1402#issuecomment-1607328925 explains why this feature is useful. While it is nice to have assessment notes #7093, their full potential only shows once there is an easy way to see which assessments have such a note.

### Description

A new column is added to the scores/submissions table, which contains little speech bubbles indicating the presence of an internal assessment note for an entry.


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise with manual assessment

1. Log in to Artemis
2. Follow the steps from #7093 to create a programming exercise and add an assessment note
3. Go to the submissions overview of the exercise
4. There should be a speech bubble in the table for the exercise you just created a note for

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots
